pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -77,10 +77,18 @@ def barrier(self, name: Optional[str] = None) -> None:
     def broadcast(self, obj: object, src: int = 0) -> object:
         """Broadcasts an object to all processes"""
 
+    @abstractmethod
+    def all_gather(self, tensor: torch.Tensor, group: Optional[Any] = None, sync_grads: bool = False) -> torch.Tensor:
+        """Perform a all_gather on all processes """
+
     def reduce_early_stopping_decision(self, should_stop: bool) -> bool:
-        """Reduce the early stopping decision across all possibly spawned processes"""
+        """Reduce the early stopping decision across all processes"""
         return should_stop
 
+    def reduce_model_checkpoint_decision(self, should_update_best_and_save: bool) -> bool:
+        """Reduce the model checkpoint decision across all processes"""
+        return should_update_best_and_save
+
```

Comment:
Is there any case where `reduce_early_stopping_decision` and `reduce_model_checkpoint_decision` reduction logic is different? Should we just have `reduce_decision`?

Reply:
Removed the logic.